CMakeLists.txt

```diff
@@ -202,51 +220,72 @@ endif (ENABLE_CLANG_AS_LIB)
 ##             TEST APP                      ##
 ###############################################
 
+## TestJit.cpp
 #----- Sources
-set ( VC_TEST_APP_SRC Test.cpp )
+# It does make no sense building it if there is no LLVM, given that the JIT implementation relies on LLVM. 
+if (ENABLE_CLANG_AS_LIB)
+  set (VC_TESTJIT_APP_SRC TestJit.cpp)
+  set (VC_EXEJIT_NAME libVC_testJit)
+  add_executable(${VC_EXEJIT_NAME} ${VC_TESTJIT_APP_SRC})
 
-set ( VC_EXE_NAME libVC_test )
+  target_link_libraries(${VC_EXEJIT_NAME} ${VC_LIB_NAME} ${VC_LIB_DEPS})
 
-if (ENABLE_CLANG_AS_LIB)
-  set_source_files_properties(${VC_TEST_APP_SRC} PROPERTIES
-                              COMPILE_FLAGS "-flto -DHAVE_CLANG_AS_LIB -DFORCED_PATH_TO_TEST=\\\"${CMAKE_CURRENT_SOURCE_DIR}/test_code\\\""
-                              )
-else (ENABLE_CLANG_AS_LIB)
-  set_source_files_properties(${VC_TEST_APP_SRC} PROPERTIES
-                              COMPILE_FLAGS "-flto -DFORCED_PATH_TO_TEST=\\\"${CMAKE_CURRENT_SOURCE_DIR}/test_code\\\""
-                              )
+  target_compile_definitions(${VC_EXEJIT_NAME} PRIVATE -DHAVE_CLANG_AS_LIB)
+  target_compile_definitions(${VC_EXEJIT_NAME} PRIVATE -DFORCED_PATH_TO_TEST="${CMAKE_CURRENT_SOURCE_DIR}/test_code")
+  if ( CMAKE_BUILD_TYPE MATCHES DEBUG )
+    target_compile_definitions(${VC_EXEJIT_NAME} PRIVATE -DVC_DEBUG)
+  endif( CMAKE_BUILD_TYPE MATCHES DEBUG )
 endif (ENABLE_CLANG_AS_LIB)
 
+## Test.cpp
+#----- Sources
+set ( VC_TEST_APP_SRC Test.cpp )
+set ( VC_EXE_NAME libVC_test )
 add_executable(${VC_EXE_NAME} ${VC_TEST_APP_SRC})
 
 target_link_libraries(${VC_EXE_NAME} ${VC_LIB_NAME} ${VC_LIB_DEPS})
 
+target_compile_definitions(${VC_EXE_NAME} PRIVATE -DFORCED_PATH_TO_TEST="${CMAKE_CURRENT_SOURCE_DIR}/test_code")
+target_compile_definitions(${VC_EXE_NAME} PRIVATE -DLLVM_TOOLS_BINARY_DIR="${LLVM_TOOLS_BINARY_DIR}")
+
+if (ENABLE_CLANG_AS_LIB)
+  target_compile_definitions(${VC_EXE_NAME} PRIVATE -DHAVE_CLANG_AS_LIB)
+endif (ENABLE_CLANG_AS_LIB)
+if ( CMAKE_BUILD_TYPE MATCHES DEBUG )
+  target_compile_definitions(${VC_EXE_NAME} PRIVATE -DVC_DEBUG)
+endif ( CMAKE_BUILD_TYPE MATCHES DEBUG )
```

Comment:
Can you please uniform across the file the formatting?
Having / not having a blank space between the `(` and the condition should be uniformed. No preference on which style to pick.